### PR TITLE
GCU portchannel interface test 

### PR DESCRIPTION
**Pull request by wen587**
### Description of PR

Summary: Testcase of portchannel interface for generic updater apply-patch
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911

#### What is the motivation for this PR?
[End to End test support for Generic Updater apply-patch](https://github.com/ghooo/SONiC/blob/c1f3f3b5427d0cafb3defd93df8b906a26fcee8a/doc/config-generic-update-rollback/SONiC_Generic_Config_Update_and_Rollback_Design.md#91-unit-tests-for-apply-patch)
This PR is to verify the usage of 'config apply-patch' works on portchannel interface
#### How did you do it?
First we setup clean portchannel interface env. Then make some config apply change. And check if the portchannel  interfaces changed as expected.

#### How did you verify/test it?
Run test of sonic-mgmt/tests/generic_config_updater/test_portchannel_interface.py on KVM

### CODE REVIEW

File: tests/generic_config_updater/test_portchannel_interface.py [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+#     "PortChannel0003": {},
+#     "PortChannel0003|10.0.0.60/31": {},
+#     "PortChannel0003|FC00::79/126": {},
+#     "PortChannel0004": {},
+#     "PortChannel0004|10.0.0.62/31": {},
+#     "PortChannel0004|FC00::7D/126": {}
+# }
+
+pytestmark = [
+    pytest.mark.topology('t0'),
```
Comment (qiluo-msft) [inline tests/generic_config_updater/test_portchannel_interface.py:27]: >t0 [](http://example.com/codeflow?start=26&length=2)

also test on t1-lag?

File: tests/generic_config_updater/test_portchannel_interface.py [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+    logger.info("json patch {}".format(json_patch))
+
+    tmpfile = generate_tmpfile(duthost)
+    logger.info("tmpfile {}".format(tmpfile))
+
+    try:
+        output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+        expect_op_success(duthost, output)
+
+        verify_attr_change(duthost, name, attr, value)
```
Comment (qiluo-msft) [inline tests/generic_config_updater/test_portchannel_interface.py:324]: >verify_attr_change [](http://example.com/codeflow?start=8&length=18)

For end-to-end test, we could also verify that teamd processes (per port channel) are running without crash.

File: tests/generic_config_updater/test_portchannel_interface.py (hunk 1 above)
Comment (wen587) [reply tests/generic_config_updater/test_portchannel_interface.py:27]: Can you find me a t1-lag topo? I can check if the test can be adopted.

File: tests/generic_config_updater/test_portchannel_interface.py (hunk 2 above)
Comment (wen587) [reply tests/generic_config_updater/test_portchannel_interface.py:324]: How about checking portchannel pid?
docker exec -it teamd test -f /var/run/teamd/PortChannel0001.pid

Comment (qiluo-msft) [reply tests/generic_config_updater/test_portchannel_interface.py:324]: Checking process such as command `ps` is better. There is risk that the process crashed but the pid file is still there.

Comment (wen587) [reply tests/generic_config_updater/test_portchannel_interface.py:335]: Changed to `state dump` verification

File: tests/generic_config_updater/test_portchannel_interface.py (hunk 1 above)
Comment (wen587) [reply tests/generic_config_updater/test_portchannel_interface.py:27]: Maybe add t1-lag in another PR. 
In this testcase, we are actually modifying the interface, like ip and attributes(mtu, min-link). Topo should not make any difference right?

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**